cachi2/core/package_managers/pip.py

```diff
@@ -1734,78 +1737,79 @@ def _process_package_distributions(
+        }
 
-    sdists = [
-        DistributionPackageInfo(
+        dpi = DistributionPackageInfo(
             name,
             version,
-            "sdist",
-            pip_deps_dir.join_within_root(p.filename).path,
-            p.url,
-            p.is_yanked,
+            cast(str, package.package_type),
+            pip_deps_dir.join_within_root(package.filename).path,
+            package.url,
+            package.is_yanked,
+            pypi_checksums,
+            user_checksums,
         )
-        for p in packages
-        if p.version
-        and canonicalize_version(p.version) == normalized_version
-        and p.package_type == "sdist"
-    ]
 
-    if not sdists:
-        raise PackageRejected(
-            f"No sdists found for package {name}=={version}",
-            solution=(
-                "It seems that this version does not exist or isn't published as a sdist "
-                "(a zip or a tarball).\n"
-                "You may be able to specify the dependency directly via a URL instead, "
-                "for example the tarball for a GitHub release."
-            ),
-            docs=PIP_NO_SDIST_DOC,
-        )
+        if dpi.package_type == "sdist":
+            sdists.append(dpi)
+        else:
+            if dpi.should_download_wheel():
+                wheels.append(dpi)
+            else:
+                log.info("Filtering out %s due to checksum mismatch", package.filename)
 
-    best_sdist = max(sdists, key=_sdist_preference)
-    if best_sdist.is_yanked:
-        raise PackageRejected(
-            f"All sdists for package {name}=={version} are yanked",
-            solution=(
-                f"Please update the {name} version in your requirements file.\n"
-                "Usually, when a version gets yanked from PyPI, there will already "
-                "be a fixed version available.\n"
-                "Otherwise, you may need to pin to the previous version."
-            ),
-        )
+    if len(sdists) != 0:
+        best_sdist = max(sdists, key=_sdist_preference)
+        if best_sdist.is_yanked:
+            raise PackageRejected(
+                f"All sdists for package {name}=={version} are yanked",
+                solution=(
+                    f"Please update the {name} version in your requirements file.\n"
+                    "Usually, when a version gets yanked from PyPI, there will already "
+                    "be a fixed version available.\n"
+                    "Otherwise, you may need to pin to the previous version."
+                ),
+            )
+    else:
+        log.warning("No source distributions found for package %s==%s", name, version)
+        best_sdist = None
+
+        if len(wheels) == 0:
+            if allow_binary:
+                solution = (
+                    "Please check that the package exists on PyPI or that the name and version are correct.\n"
+                    "Consider using option allow_binary to allow wheel distributions."
```

Comment:
This is in the `allow_binary == True` context, so the suggestion to "Consider using allow_binary" doesn't apply